commit da2637dd8a23
Author: David Keeler <dkeeler@mozilla.com>
Date:   Wed Oct 25 09:54:13 2017 -0700

    Bug 1411458 - Confirm we actually have a PKCS#7 signedData content info. r=jcj, a=ritu
    
    MozReview-Commit-ID: GKfL1C0EPWt
    
    --HG--
    extra : source : 22a772cfa62ae6908751b1cdd98a9234f2ea6105
---
 security/manager/ssl/nsDataSignatureVerifier.cpp | 6 ++++++
 security/nss.symbols                             | 1 +
 2 files changed, 7 insertions(+)

diff --git security/manager/ssl/nsDataSignatureVerifier.cpp security/manager/ssl/nsDataSignatureVerifier.cpp
index f78740365e1a..2527431dd9ae 100644
--- security/manager/ssl/nsDataSignatureVerifier.cpp
+++ security/manager/ssl/nsDataSignatureVerifier.cpp
@@ -170,6 +170,12 @@ VerifyCMSDetachedSignatureIncludingCertificate(
     return NS_ERROR_CMS_VERIFY_NO_CONTENT_INFO;
   }
 
+  // We're expecting this to be a PKCS#7 signedData content info.
+  if (NSS_CMSContentInfo_GetContentTypeTag(cinfo)
+        != SEC_OID_PKCS7_SIGNED_DATA) {
+    return NS_ERROR_CMS_VERIFY_NO_CONTENT_INFO;
+  }
+
   // signedData is non-owning
   NSSCMSSignedData* signedData =
     static_cast<NSSCMSSignedData*>(NSS_CMSContentInfo_GetContent(cinfo));
diff --git security/nss.symbols security/nss.symbols
index 626fc64d96f9..517a577a8de4 100644
--- security/nss.symbols
+++ security/nss.symbols
@@ -181,6 +181,7 @@ HASH_ResultLenByOidTag
 HASH_Update
 NSSBase64_EncodeItem_Util
 NSS_CMSContentInfo_GetContent
+NSS_CMSContentInfo_GetContentTypeTag
 NSS_CMSContentInfo_SetContent_Data
 NSS_CMSContentInfo_SetContent_EnvelopedData
 NSS_CMSContentInfo_SetContent_SignedData
